src/Services/DocumentTemplates/DocumentTemplateService.php

```diff
@@ -12,12 +12,331 @@
+    }
+
+    /**
+     * @param $profile
+     * @return array
+     */
+    public function getTemplateListByProfile($profile): array
+    {
+        $rtn = sqlStatement('SELECT `template_id`, `category` FROM `document_template_profiles` WHERE `profile` = ? AND `template_id` > 0', array($profile));
+        $profile_list = array();
+        while ($row = sqlFetchArray($rtn)) {
+            $profile_list[$row['category']][] = $this->fetchTemplate($row['template_id']);
+        }
+        return $profile_list;
+    }
+
+    /**
+     * @param      $id
+     * @param null $template_name
+     * @return array|false|null
+     */
+    public function fetchTemplate($id, $template_name = null)
+    {
+        $return = null;
+        if (!empty($id)) {
+            $return = sqlQuery('SELECT * FROM `document_templates` WHERE `id` = ?', array($id));
+        } elseif (!empty($template_name)) {
+            $return = sqlQuery('SELECT * FROM `document_templates` WHERE `template_name` = ?', array($template_name));
+        }
+        return $return;
+    }
+
+    /**
+     * @param $profile_groups
+     * @return bool
+     */
+    public function savePatientGroupsByProfile($profile_groups): bool
+    {
+        sqlStatementNoLog('SET autocommit=0');
+        sqlStatementNoLog('START TRANSACTION');
+
+        try {
+            sqlQuery('DELETE From `document_template_profiles` WHERE `template_id` = 0');
+            $sql = 'INSERT INTO `document_template_profiles` (`id`, `template_id`, `profile`, `template_name`, `category`, `provider`, `modified_date`, `member_of`, `active`) VALUES (NULL, 0, ?, "", "Group", ?, current_timestamp(), ?, ?)';
+
+            foreach ($profile_groups as $profile => $groups) {
+                foreach ($groups as $group) {
+                    $rtn = sqlInsert($sql, array($profile, $_SESSION['authUserID'] ?? null, $group['group'] ?? '', $group['active']));
+                }
+            }
+        } catch (Exception $e) {
+            throw new RuntimeException($e->getMessage(), $e->getCode(), $e);
+        }
+        sqlStatementNoLog('COMMIT');
+        sqlStatementNoLog('SET autocommit=1');
+
+        return $rtn ?? 0;
+    }
+
+    /**
+     * @param $patients
+     * @return bool
+     */
+    public function updateGroupsInPatients($patients): bool
+    {
+        sqlStatementNoLog('SET autocommit=0');
+        sqlStatementNoLog('START TRANSACTION');
+        try {
+            $rtn = sqlQuery('UPDATE `patient_data` SET `patient_groups` = ? WHERE `pid` > ?', array(null, 0));
+            foreach ($patients as $id => $groups) {
+                $rtn = sqlQuery('UPDATE `patient_data` SET `patient_groups` = ? WHERE `pid` = ?', array($groups, $id));
+            }
+            sqlStatementNoLog('COMMIT');
+            sqlStatementNoLog('SET autocommit=1');
+        } catch (Exception $e) {
+            throw new RuntimeException($e->getMessage(), $e->getCode(), $e);
+        }
+        return !$rtn;
```

Comment:
missing the : 
```
        sqlStatementNoLog('COMMIT');
        sqlStatementNoLog('SET autocommit=1');
```

Reply:
oh wait, disregard, it is there. it is just within the try statement whereas the savePatientGroupsByProfile function had them outside (ie. i was not looking in the right place :) )

Reply:
though will work, still should be outside the try, good catch